pkg/sdk/testint/tables_integration_test.go

```diff
@@ -0,0 +1,949 @@
+		}
+		oldConstraintName := "OUT_OF_LINE_CONSTRAINT"
+		outOfLineConstraint := sdk.NewOutOfLineConstraintRequest(oldConstraintName, sdk.ColumnConstraintTypePrimaryKey).WithColumns([]string{"COLUMN_1"})
+
+		err := client.Tables.Create(ctx, sdk.NewCreateTableRequest(id, columns).WithOutOfLineConstraint(outOfLineConstraint))
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		newConstraintName := "NEW_OUT_OF_LINE_CONSTRAINT_NAME"
+		alterRequest := sdk.NewAlterTableRequest(id).
+			WithConstraintAction(sdk.NewTableConstraintActionRequest().
+				WithRename(sdk.NewTableConstraintRenameActionRequest().
+					WithOldName(oldConstraintName).
+					WithNewName(newConstraintName)))
+		err = client.Tables.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+	})
+
+	// TODO: check altered constraint
+	t.Run("alter constraint: alter", func(t *testing.T) {
+		name := random.String()
+		id := sdk.NewSchemaObjectIdentifier(database.Name, schema.Name, name)
+		columns := []sdk.TableColumnRequest{
+			*sdk.NewTableColumnRequest("COLUMN_1", sdk.DataTypeVARCHAR),
+			*sdk.NewTableColumnRequest("COLUMN_2", sdk.DataTypeVARCHAR),
+		}
+		constraintName := "OUT_OF_LINE_CONSTRAINT"
+		outOfLineConstraint := sdk.NewOutOfLineConstraintRequest(constraintName, sdk.ColumnConstraintTypePrimaryKey).WithColumns([]string{"COLUMN_1"})
+
+		err := client.Tables.Create(ctx, sdk.NewCreateTableRequest(id, columns).WithOutOfLineConstraint(outOfLineConstraint))
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		alterRequest := sdk.NewAlterTableRequest(id).
+			WithConstraintAction(sdk.NewTableConstraintActionRequest().WithAlter(sdk.NewTableConstraintAlterActionRequest().WithConstraintName(sdk.String(constraintName)).WithEnforced(sdk.Bool(true))))
+		err = client.Tables.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+	})
+
+	// TODO: check dropped constraint
+	t.Run("alter constraint: drop", func(t *testing.T) {
+		name := random.String()
+		id := sdk.NewSchemaObjectIdentifier(database.Name, schema.Name, name)
+		columns := []sdk.TableColumnRequest{
+			*sdk.NewTableColumnRequest("COLUMN_1", sdk.DataTypeVARCHAR),
+			*sdk.NewTableColumnRequest("COLUMN_2", sdk.DataTypeVARCHAR),
+		}
+		constraintName := "OUT_OF_LINE_CONSTRAINT"
+		outOfLineConstraint := sdk.NewOutOfLineConstraintRequest(constraintName, sdk.ColumnConstraintTypePrimaryKey).WithColumns([]string{"COLUMN_1"})
+
+		err := client.Tables.Create(ctx, sdk.NewCreateTableRequest(id, columns).WithOutOfLineConstraint(outOfLineConstraint))
+		require.NoError(t, err)
+		t.Cleanup(cleanupTableProvider(id))
+
+		alterRequest := sdk.NewAlterTableRequest(id).
+			WithConstraintAction(sdk.NewTableConstraintActionRequest().WithDrop(sdk.NewTableConstraintDropActionRequest().WithConstraintName(sdk.String(constraintName))))
+		err = client.Tables.Alter(ctx, alterRequest)
+		require.NoError(t, err)
+	})
+
+	t.Run("external table: add", func(t *testing.T) {
```

Comment:
I'm guessing this should also be available in the external table alter, right ?

Reply:
Left comment on `ExternalTables`.